### PR TITLE
Attempt to improve docs

### PR DESCRIPTION
**Pull request by jasonswett**
Based on my recent post [The problem that ViewComponent solves for me](https://www.codewithjason.com/the-problem-that-viewcomponent-solves-for-me/), Joel asked me if I might want to take a swing at improving the ViewComponent landing page docs.

Here's my attempt!

### CODE REVIEW

**Comment by Spone** [comment]
I read your blog post with interest and I'm happy that these insights are used to improve the docs!

I'm wondering if the "What" could stay at the top, so you see code samples early (personally, I find it helpful to see the code to understand what a library provides), then you get to understand the benefits.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ title: Changelog
 
 ## main
 
+## 2.53.0
```
Comment (Spone) [inline docs/CHANGELOG.md:10]: Can you remove this heading? It will be added as part of the release process.

Comment (jasonswett) [reply docs/CHANGELOG.md]: Done!

**Comment by jasonswett** [comment]
> I'm wondering if the "What" could stay at the top, so you see code samples early (personally, I find it helpful to see the code to understand what a library provides), then you get to understand the benefits.

Done!

**Comment by joelhawksley** [comment]
@jasonswett sorry for the delay in getting to your PR. I _really_ appreciate what you've added here! I'm going to give this a read and try to condense it down for the sake of keeping the homepage approachable ❤️

**Comment by jasonswett** [comment]
No problem at all! I appreciate the opportunity to contribute.

**Comment by joelhawksley** [comment]
Closing in favor of https://github.com/github/view_component/pull/1331 ❤️

**Pull request closed without merging**